### ISSUE TITLE
Switch to a C based JSON parser

### ISSUE THREAD

**Issue by ZeroIntensity** [ZeroIntensity/view.py#193]
### Description:

As of now, we use [uJSON](https://github.com/ultrajson/ultrajson) for JSON parsing. Unfortunately, they've basically marked it as security-only, which isn't a great look for view.py. The obvious, drop-in replacement for now would be [ORJson](https://github.com/ijl/orjson/), but that might be very bad in the extreme. Namely, this would force us to set our memory leak limit very high in our unit testing suite, due to an overallocation issue on their end.

So, what's the solution? I'm thinking that [cJSON](https://github.com/DaveGamble/cJSON) is probably our best bet, or if I hate myself and want to cross-compile view.py with C++, then [RapidJson](https://github.com/Tencent/rapidjson). This will unfortunately require a lot of changes to the parser implementation.

It might be worth trying to migrate all our current and future parsers to C, depending on what libraries exist.

**Comment by sanbei011** [comment]
ORJson maintainer is really careless, there is a memory leak bug that has not moved for two months, and the prs submitted by people are always automatically closed

**Comment by ZeroIntensity** [comment]
> ORJson maintainer is really careless, there is a memory leak bug that has not moved for two months, and the prs submitted by people are always automatically closed

Correct, the point of this issue is to *not* use ORJson, specifically for that reason.

**Comment by ZeroIntensity** [comment]
A quick note regarding the ORJson maintainer: I was a bit harsh in my original issue description. I said the following (which I've now edited out):

> but overall, I simply don't really trust the ORJson maintainer. They've made only a couple hundred commits in the past few years, have *never* merged a PR on ORJson, and tend to let issues go stale and automatically closed.

I was wrong for this comment, that's my fault -- open source is hard, and that's not the ORJson maintainer's fault.
